torch/_weights_only_unpickler.py

```diff
@@ -59,11 +61,32 @@
     UnpicklingError,
 )
 from struct import unpack
-from sys import maxsize
-from typing import Any, Dict, List
+from sys import maxsize, modules
+from typing import Any, Callable, Dict, List
 
 import torch
 
+_marked_safe_globals_list: List[Callable] = []
+
+
+def _mark_safe_globals(safe_globals: List[Callable]):
```

Comment:
From security point-of-view I would prefer the list of safe globals to be an explicit argument to torch.load rather than a global state, as relying on global state is prone to hard-to-track security violations (somebody can mark something as safe during import of third-party library and then all users of torch will assume it's safe to load `.pth`, while in reality it is not, because someplace else global state was modified to allow that

Reply:
Sorry for not updating here. Updating this to reflect the latest view:

Passing the allowed globals as a kwarg to to `torch.load` might be prohibitive as users might not have access to the `torch.load` call (e.g. the end-user facing API might be something like `from_pretrained`. As such, having such global state is a tradeoff we might need to make. We will provide an API to get the current state of the global list/remove things from the global list.

I will update the PR to do the above if we agree on the approach in the google doc.